packages/algob/src/lib/algo-operator.ts

```diff
@@ -54,18 +62,103 @@ export class AlgoOperatorImpl implements AlgoOperator {
     }
   };
 
+  getTxFee (params: algosdk.SuggestedParams, txSize: number): number {
+    if (params.flatFee) {
+      return Math.max(ALGORAND_MIN_TX_FEE, params.fee);
+    }
+    return Math.max(ALGORAND_MIN_TX_FEE, txSize);
+  }
+
+  getUsableAccBalance (accoutInfo: algosdk.AccountInfo): number {
+    // Extracted from interacting with Algorand node:
+    // 7 opted-in assets require to have 800000 micro algos (frozen in account).
+    // 11 assets require 1200000.
+    return accoutInfo.amount - (accoutInfo.assets.length + 1) * ALGORAND_ASA_OWNERSHIP_COST;
+  }
+
+  getOptInTxSize (params: algosdk.SuggestedParams, accounts: Accounts): number {
+    const randomAccount = accounts.values().next().value;
+    // assetID can't be known before ASA creation
+    // it shouldn't be easy to find out the latest asset ID
+    // In original source code it's uint64:
+    // https://github.com/algorand/go-algorand/blob/1424855ad2b5f6755ff3feba7e419ee06f2493da/data/basics/userBalance.go#L278
+    const assetID = Number.MAX_SAFE_INTEGER; // not 64 bits but 55 bits should be enough
+    const sampleASAOptInTX = tx.makeASAOptInTx(randomAccount.addr, assetID, params);
+    const rawSignedTxn = sampleASAOptInTX.signTxn(randomAccount.sk);
+    return rawSignedTxn.length;
+  }
+
+  async optInToASA (
+    asaName: string, assetIndex: number, account: Account, params: algosdk.SuggestedParams
+  ): Promise<void> {
+    console.log(`ASA ${account.name} opt-in for for ASA ${asaName}`);
+    const sampleASAOptInTX = tx.makeASAOptInTx(account.addr, assetIndex, params);
+    const rawSignedTxn = sampleASAOptInTX.signTxn(account.sk);
+    const txInfo = await this.algodClient.sendRawTransaction(rawSignedTxn).do();
+    await this.waitForConfirmation(txInfo.txId);
+  }
```

Comment:
This function must be also exported to user through deployer, however the deployer version should take as a parameter `TxParams` instead of `algosdk.SuggestedParams`.

Reply:
This function also takes assetID which is too verbose as well.
I'll make a new function that will prepare the parameters for this one and won't touch this function.

Reply:
I'll use this as a signature for `AlgoOperator`:
`optInToASA (asaName: string, assetIndex: number, account: Account, params: DeploymentFlags): Promise<void>`
And this for `Deployer`:
`optInToASA (asaName: string, account: Account, params: DeploymentFlags): Promise<void>`

It will also need to check that `asaName` has been saved in CP

Reply:
Deployer's signature changed to
`optInToASA (name: string, accountName: string, flags: ASADeploymentFlags): Promise<void>`
`accountName` is better than `Account`

Reply:
Example updated. `tesla` will do the manual opt-in.

Reply:
Why `DeploymentFlags` instead of `TxParams` (isn't the base interface enough)?

Reply:
It looks like I have an old code. `DeploymentFlags` was the old name of `TxParams`